holoviews/core/decollate.py

```diff
@@ -0,0 +1,255 @@
+import param
+from .operation import OperationCallable
+from .. import (
+    Layout, DynamicMap, Element, Callable, Overlay, GridSpace, NdOverlay, HoloMap
+)
+from . import ViewableTree
+from collections import namedtuple
+from ..streams import Stream, Derived
+
+from ..plotting.util import initialize_dynamic
+
+HoloViewsExpr = namedtuple("HoloviewsExpr", ["fn", "args", "kwargs"])
+StreamIndex = namedtuple("StreamIndex", ["index"])
+KDimIndex = namedtuple("KDim", ["index"])
+
+
+def to_expr_extract_streams(
```

Comment:
Bit of a monster function but that seems unavoidable given what it is trying to do. As it is simply building an expression, it might be nice to have one or two tests of just this function (which can also help make what it does more concrete).